### PR TITLE
fix: allow CartItemDto to accept empty or invalid picture/url withou…

### PR DESCRIPTION
**Pull request by joyet-simon**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-4023/release-bug-invalid-format-url-error)
Product URLs and picture URLs are cosmetic/secondary data. They should never block a payment creation. When a product has no image, the payment should proceed normally.

#### CartItemDto.php

- Removed pictureUrl from constructor — it is not mandatory data

- setPictureUrl() and setUrl() now accept nullable values and silently ignore invalid/empty/null URLs instead of throwing InvalidArgumentException

- Initialized all nullable properties with default null values to avoid typed property access errors

#### CartItemDtoTest.php

- Updated constructor calls (removed pictureUrl argument)

- Added tests: null pictureUrl, empty pictureUrl, no pictureUrl, invalid URLs silently ignored

#### PaymentDtoTest.php

- Moved pictureUrl from constructor to setPictureUrl() call

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: src/Client/Application/DTO/CartItemDto.php [hunk 1]
```diff
@@ -5,24 +5,22 @@
 use InvalidArgumentException;
 
 class CartItemDto implements DtoInterface {
-    private ?string $sku;
-    private ?string $title;
+    private ?string $sku = null;
+    private ?string $title = null;
```
Comment (Benjamin-Freoua-Alma) [inline src/Client/Application/DTO/CartItemDto.php]: Title can be nullable ?!

**Review by Benjamin-Freoua-Alma — CHANGES_REQUESTED** [review]

File: src/Client/Application/DTO/CartItemDto.php (hunk 1 above)
Comment (joyet-simon) [reply src/Client/Application/DTO/CartItemDto.php]: Done

File: src/Client/Application/DTO/CartItemDto.php
Comment (Benjamin-Freoua-Alma) [inline src/Client/Application/DTO/CartItemDto.php:37]: If it's not nullable we need to add verification if title is empty like
```
if (empty($title)) {
            throw new InvalidArgumentException("title can't be empty.");
        }
```

File: tests/Client/Unit/Application/DTO/CartItemDtoTest.php [hunk 2]
```diff
@@ -39,30 +36,51 @@ public function testCartItemDto()
...
-        (new CartItemDto(1, 25, 'https://example.com/image.jpg'))->setQuantity(0);
+        new CartItemDto(0, 25, 'My product');
     }
 
     public function testInvalidUnitPrice()
     {
         $this->expectException(InvalidArgumentException::class);
-        (new CartItemDto(1, 25, 'https://example.com/image.jpg'))->setUnitPrice(-1);
+        (new CartItemDto(1, 25, 'My product'))->setUnitPrice(-1);
     }
```
Comment (Benjamin-Freoua-Alma) [inline tests/Client/Unit/Application/DTO/CartItemDtoTest.php:47]: Maybe add test to check the empty value in Title

**Review by Benjamin-Freoua-Alma — CHANGES_REQUESTED** [review]

**Review by Benjamin-Freoua-Alma — CHANGES_REQUESTED** [review]
Check title empty and add test

**Review by Benjamin-Freoua-Alma — APPROVED** [review]
